lib/kubernetes-deploy/kubectl.rb

```diff
@@ -20,26 +21,35 @@ def initialize(namespace:, context:, logger:, log_failure_by_default:, default_t
       raise ArgumentError, "context is required" if context.blank?
     end
 
-    def run(*args, log_failure: nil, use_context: true, use_namespace: true, raise_if_not_found: false)
+    def run(*args, log_failure: nil, use_context: true, use_namespace: true, raise_if_not_found: false, attempts: 1)
       log_failure = @log_failure_by_default if log_failure.nil?
 
       args = args.unshift("kubectl")
       args.push("--namespace=#{@namespace}") if use_namespace
       args.push("--context=#{@context}")     if use_context
       args.push("--request-timeout=#{@default_timeout}") if @default_timeout
-
-      @logger.debug Shellwords.join(args)
-      out, err, st = Open3.capture3(*args)
-      @logger.debug(out.shellescape) unless output_is_sensitive?
-
-      unless st.success?
-        if log_failure
-          @logger.warn("The following command failed: #{Shellwords.join(args)}")
-          @logger.warn(err) unless output_is_sensitive?
-        end
-
-        if raise_if_not_found && err.match(NOT_FOUND_ERROR_TEXT)
-          raise ResourceNotFoundError, err
+      out, err, st = nil
+
+      (1..attempts).to_a.each do |attempt|
+        @logger.debug "Running command (attempt #{attempt}): #{args.join(' ')}"
+        out, err, st = Open3.capture3(*args)
+        @logger.debug("Kubectl out: " + out.gsub(/\s+/, ' ')) unless output_is_sensitive?
+
+        if st.success?
+          break
+        else
+          if log_failure
+            @logger.warn("The following command failed: #{Shellwords.join(args)}")
+            @logger.warn(err) unless output_is_sensitive?
+          end
+
+          if err.match(NOT_FOUND_ERROR_TEXT)
+            raise(ResourceNotFoundError, err) if raise_if_not_found
+          else
+            @logger.debug("Kubectl err: #{err}") unless output_is_sensitive?
+            ::StatsD.increment('kubectl.error', 1, tags: { context: @context, namespace: @namespace, cmd: args[1] })
+          end
+          sleep RETRY_DELAY unless attempt == attempts
```

Comment:
Any reason why you decided not to have increasing back off on each attempt?

Reply:
Not really. Just did the simplest thing to start. Another simple option could be `sleep attempt` though (so an additional second each time).

Reply:
`sleep attempt` works for me.